server/controllers/eventsController.ts

```diff
@@ -0,0 +1,137 @@
+import { Request, Response } from 'express';
+import { Event } from 'server/models/Event';
+import { PostgresErrorCodes } from 'server/util/PostgresErrorConstants';
+
+// The whole model is a json response, fix that if there's some sensitive data here
+
+export default {
+  async index(req: Request, res: Response) {
+    const { chapterId } = req.params;
+
+    const events = await Event.find({
+      where: { chapter: chapterId },
+    });
+
+    res.json(events);
+  },
+
+  async show(req: Request, res: Response) {
+    const { id, chapterId } = req.params;
+
+    const event = await Event.findOne({
+      where: {
+        id: parseInt(id),
+        chapter: chapterId,
+      },
+    });
+
+    if (event) {
+      res.json(event);
+    } else {
+      res.status(404).json({ error: "Can't find event" });
+    }
+  },
+
+  async create(req: Request, res: Response) {
+    const {
+      name,
+      chapter,
+      description,
+      capacity,
+      venue,
+      canceled,
+      start_at,
+      ends_at,
+    } = req.body;
+
+    const event = new Event({
+      name,
+      chapter,
+      description,
+      capacity,
+      venue,
+      canceled,
+      start_at,
+      ends_at,
+    });
+
+    try {
+      await event.save();
+      res.status(201).json(event);
+    } catch (e) {
+      if (e.code === PostgresErrorCodes.FOREIGN_KEY_VIOLATION) {
+        if (e.detail.includes('venue')) {
+          return res.status(404).json({ message: 'venue not found' });
+        }
+        if (e.detail.includes('chapter')) {
+          return res.status(404).json({ message: 'chapter not found' });
+        }
+      }
+
+      res.status(500).json({ error: e });
```

Comment:
Yeah, we should do that instead of using `404`, but also keep the 500 error on line 71

Reply:
according to rest API conventions, we should give 500 when there is a server-side error, but here is the fault is from the client-side so I think we should not give 500 as a response code.
Ref: https://restfulapi.net/http-status-codes/

Reply:
@vaibhavsingh97 What makes you think it is the fault of the client?

Reply:
Client fault as in, the client didn't send all the details required by API. Here server didn't break, it's the data which is missing

Reply:
It's a generic catch all response that is surely not a foreign key violation. How can you be sure it's the data which is missing?

Reply:
`res.status(500).json({ error: e })`
Could catch client side errors or even unique constraint errors any other error we are not aware of. why don't we have validation in place to catch all client side errors instead.

Reply:
Could we just have this then? Return 400 for client side errors, and a generic 500 for any other errors?

```typescript
if (e.code === PostgresErrorCodes.FOREIGN_KEY_VIOLATION) {
  if (e.detail.includes('venue')) {
    return res.status(400).json({ message: 'venue not found' });
  }
  if (e.detail.includes('chapter')) {
    return res.status(400).json({ message: 'chapter not found' });
  }
}

res.status(500).json({ error: e });
```

Reply:
I still don't understand why any of the current code needs to change. It's fine.

Reply:
I agree with you @AryanJ-NYC, only change 404 -> 400, because it's a client error and not a not found

Reply:
@AryanJ-NYC @ankorGH will explain with example later today, but 5xx error generally means that server has taken responsibility. Also, `res.status(500).json({ error: e }) ` will return everything including the SQL query it ran for the request. Are we fine with this?

Reply:
I fully understand what a 500 error is. What we're trying to communicate is that this _DOES_ fall in the realm of a server error.

Reply:
+1 @vaibhavsingh97. 

@AryanJ-NYC why this fall in the realm of a server error? If you think this should go as a 5xx then please give me an example of a client-side error that may creep when we create an event.

Reply:
I changed this to 400 for now, we can open a new issue to discuss what we're going to use. But I think this is fine for now

Reply:
Yes, we can fix this in another PR after discussion. Approving

Reply:
@vkWeb I don't follow your logic. If I think thing A, I have to give an example of thing B? Why the pop quiz?

Reply:
@AryanJ-NYC I asked that question to make you think of situations when a client-side error may creep. 

I checked the models. We aren't allowing NULL values for event name and capacity. So shouldn't we catch `NOT_NULL_VIOLATION` for them?

Reply:
Instead of asking questions to try and make me think, wouldn't it be easier to just explicitly say: "Hey, we're missing a catch for the `NOT_NULL_VIOLATION`? 

I'm done arguing this, FWIW.